homeassistant/components/synology_dsm/__init__.py

```diff
@@ -214,32 +209,92 @@ def _async_migrator(entity_entry: entity_registry.RegistryEntry):
             entry, data={**entry.data, CONF_MAC: network.macs}
         )
 
-    # setup DataUpdateCoordinator
-    async def async_coordinator_update_data_surveillance_station():
-        """Fetch all surveillance station data from api."""
+    async def async_coordinator_update_data_cameras():
+        """Fetch all camera data from api."""
+        if not hass.data[DOMAIN][entry.unique_id][SYSTEM_LOADED]:
+            raise UpdateFailed("System not fully loaded")
+
+        if SynoSurveillanceStation.CAMERA_API_KEY not in api.dsm.apis:
+            return None
+
         surveillance_station = api.surveillance_station
+
         try:
             async with async_timeout.timeout(10):
                 await hass.async_add_executor_job(surveillance_station.update)
         except SynologyDSMAPIErrorException as err:
+            _LOGGER.debug(
+                "async_coordinator_update_data_cameras() - exception: %s", err
+            )
             raise UpdateFailed(f"Error communicating with API: {err}") from err
 
-        if SynoSurveillanceStation.CAMERA_API_KEY not in api.dsm.apis:
-            return
-
         return {
             "cameras": {
                 camera.id: camera for camera in surveillance_station.get_all_cameras()
             }
         }
 
-    hass.data[DOMAIN][entry.unique_id][
-        COORDINATOR_SURVEILLANCE
-    ] = DataUpdateCoordinator(
+    async def async_coordinator_update_data_central():
+        """Fetch all device and sensor data from api."""
+        registry = await entity_registry.async_get_registry(hass)
+        entries = entity_registry.async_entries_for_config_entry(
+            registry, entry.entry_id
+        )
+        used_keys = {}
+        for ent in entries:
+            api_key = ent.unique_id.split(":")[0].split("_")[-1]
+            used_keys[api_key] = True
+        api.set_fetching_entities(used_keys)
```

Comment:
Can we avoid accessing the entity registry each update call?

Reply:
It is needed to decide, which APIs should be polled or not, based on enabled entities (_ex. if all storage related entities are disabled by user, than `dsm.storage` is no longer needed to be polled_)

Reply:
the former way was to subscribe each entity during setup by `async_added_to_hass()` into `SynoApi()._fetching_entities` but this is not possible anymore when using `CoordinatorEntity` because it already implements `async_added_to_hass()` by its own.

Reply:
You can still have `async_added_to_hass` to track of them and then call super().async_added_to_hass() to make sure the `CoordinatorEntity` version gets called.

Reply:
ah ok ... I will give it try in the upcoming days